llvm/lib/Transforms/Instrumentation/HWAddressSanitizer.cpp

```diff
@@ -1527,6 +1533,9 @@ void HWAddressSanitizer::sanitizeFunction(Function &F,
 
   NumTotalFuncs++;
   if (CSkipHotCode) {
+    if ((F.getGUID() % MaxRandomRate) < SkipInstRandomRate) {
```

Comment:
@vitalybuka probably not correct logic. PTAL.

Reply:
How is this getGUID value generated?
I wonder if selection will be shuffled after some code changes/compiler updates?
I would assume we want (1) stability with the same code/compiler for reproducibility, (2) shuffle selection on code/compiler updates to get more coverage over time.

Reply:
I guess we agreed to use `createRNG(F.getName());` like in #83471